### ISSUE TITLE
npm run zap error --no-sandbox

### ISSUE THREAD

**Issue by cheng5230** [project-chip/zap#961]
> zap@2023.3.1 zap
> node src-script/zap-start.js --logToStdout --gen ./test/gen-template/zigbee/gen-templates.json

🚀 Executing: git log -1 --format={"hash": "%H","timestamp": %ct}
🔍 Git commit: 4bcb914b8ce213a570e60888a4349cce021863f7 from Wed Mar 01 2023 03:56:16 GMT+0900 (Japan Standard Time)
🔍 Current src hash: C8b6acTn4n7+SgjnyANw0myYFAA=
🔍 Current src-shared hash: //f+yOV4+MNjwnqqpFCZgz3pjuU=
🔍 Previous src hash: C8b6acTn4n7+SgjnyANw0myYFAA=
🔍 Previous src-shared hash: //f+yOV4+MNjwnqqpFCZgz3pjuU=
👍 There were no changes to front-end code, so we don't have to rebuild the SPA.
🚀 Executing: npx tsc --build ./tsconfig.json
🚀 Executing: npx copyfiles ./src-electron/**/*.sql ./src-electron/icons/* ./dist/
🚀 Executing: npx electron --disable-seccomp-filter-sandbox /home/myshare/zap/dist/src-electron/ui/main-ui.js --logToStdout --gen ./test/gen-template/zigbee/gen-templates.json
⇝ [0302/174850.979808:FATAL:electron_main_delegate.cc(300)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
⇝ /home/myshare/zap/node_modules/electron/dist/electron exited with signal SIGTRAP
👎 Program npx exited with error code: 1
1

**Comment by jingteng25742** [comment]
Hi @cheng5230 

Are you able to reproduce the issue with release binaries version of ZAP at 
[](https://github.com/project-chip/zap/releases)

**Comment by cheng5230** [comment]
yes i use [Releases](https://github.com/project-chip/zap/releases) [v2023.03.06-nightly](https://github.com/project-chip/zap/releases/tag/v2023.03.06-nightly)

**run zap display error**

 ./zap --no-sandbox
🔧 Using state directory: /root/.zap
[7667:0309/111150.727592:ERROR:bus.cc(397)] Failed to connect to the bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
[7667:0309/111150.727948:ERROR:bus.cc(397)] Failed to connect to the bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
No protocol specified
No protocol specified
No protocol specified
No protocol specified
[7695:0309/111150.793278:ERROR:angle_platform_impl.cc(44)] Display.cpp:940 (initialize): ANGLE Display::initialize error 12289: Could not open the default X display.
[7695:0309/111150.794604:ERROR:gl_surface_egl.cc(808)] EGL Driver message (Critical) eglInitialize: Could not open the default X display.
[7695:0309/111150.795160:ERROR:gl_surface_egl.cc(1430)] eglInitialize OpenGL failed with error EGL_NOT_INITIALIZED, trying next display type
No protocol specified
[7695:0309/111150.796710:ERROR:angle_platform_impl.cc(44)] Display.cpp:940 (initialize): ANGLE Display::initialize error 12289: Could not open the default X display.
[7695:0309/111150.797220:ERROR:gl_surface_egl.cc(808)] EGL Driver message (Critical) eglInitialize: Could not open the default X display.
[7695:0309/111150.797822:ERROR:gl_surface_egl.cc(1430)] eglInitialize OpenGLES failed with error EGL_NOT_INITIALIZED
[7695:0309/111150.798042:ERROR:gl_ozone_egl.cc(20)] GLSurfaceEGL::InitializeOneOff failed.
[7695:0309/111150.804491:ERROR:viz_main_impl.cc(188)] Exiting GPU process due to errors during initialization
No protocol specified
No protocol specified
No protocol specified
No protocol specified
[7707:0309/111150.831927:ERROR:angle_platform_impl.cc(44)] DisplayVkXcb.cpp:59 (initialize): xcb_connect() failed, error 1
[7707:0309/111150.832834:ERROR:angle_platform_impl.cc(44)] Display.cpp:940 (initialize): ANGLE Display::initialize error 12289: Not initialized.
[7707:0309/111150.833582:ERROR:gl_surface_egl.cc(808)] EGL Driver message (Critical) eglInitialize: Not initialized.
[7707:0309/111150.833872:ERROR:gl_surface_egl.cc(1430)] eglInitialize SwANGLE failed with error EGL_NOT_INITIALIZED
[7707:0309/111150.834173:ERROR:gl_ozone_egl.cc(20)] GLSurfaceEGL::InitializeOneOff failed.
[7707:0309/111150.846805:ERROR:viz_main_impl.cc(188)] Exiting GPU process due to errors during initialization
No protocol specified
No protocol specified
No protocol specified
No protocol specified
[7721:0309/111150.876113:ERROR:angle_platform_impl.cc(44)] DisplayVkXcb.cpp:59 (initialize): xcb_connect() failed, error 1
[7721:0309/111150.876500:ERROR:angle_platform_impl.cc(44)] Display.cpp:940 (initialize): ANGLE Display::initialize error 12289: Not initialized.
[7721:0309/111150.876699:ERROR:gl_surface_egl.cc(808)] EGL Driver message (Critical) eglInitialize: Not initialized.
[7721:0309/111150.876822:ERROR:gl_surface_egl.cc(1430)] eglInitialize SwANGLE failed with error EGL_NOT_INITIALIZED
[7721:0309/111150.876948:ERROR:gl_ozone_egl.cc(20)] GLSurfaceEGL::InitializeOneOff failed.
[7721:0309/111150.882983:ERROR:viz_main_impl.cc(188)] Exiting GPU process due to errors during initialization
No protocol specified
No protocol specified
[7729:0309/111150.891336:ERROR:gpu_init.cc(454)] Passthrough is not supported, GL is disabled, ANGLE is 
[7729:0309/111150.891804:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.

**Comment by paulr34** [comment]
its working for me with the latest

**paregan@mac0015027 zap % npm run zap display error

> zap@0.0.0 zap
> node src-script/zap-start.js --logToStdout --gen ./test/gen-template/zigbee/gen-templates.json display error

🚀 Executing: git log -1 --format={"hash": "%H","timestamp": %ct}
🔍 Git commit: 15f96c4f997301da1174ef29156c1c7d442f5d44 from Mon Mar 27 2023 13:20:50 GMT-0400 (Eastern Daylight Time)
🔍 Current src hash: tt1kGIKwkwmZc8maXjP1W3u9tu4=
🔍 Current src-shared hash: g/WbG5bVfcehj7QTAknSkxvoR5c=
🔍 Previous src hash: tt1kGIKwkwmZc8maXjP1W3u9tu4=
🔍 Previous src-shared hash: g/WbG5bVfcehj7QTAknSkxvoR5c=
👍 There were no changes to front-end code, so we don't have to rebuild the SPA.
🚀 Executing: npx tsc --build ./tsconfig.json
🚀 Executing: npx copyfiles ./src-electron/**/*.sql ./src-electron/icons/* ./dist/
🚀 Executing: npx electron /Users/paregan/zap/dist/src-electron/ui/main-ui.js --logToStdout --gen ./test/gen-template/zigbee/gen-templates.json display error
🔧 Using state directory: /Users/paregan/.zap
{"level":40,"time":1680021934992,"pid":75175,"hostname":"mac0015027","name":"zap","msg":"Can't locate templates file: /Users/paregan/zap/display"}
{"level":40,"time":1680021934992,"pid":75175,"hostname":"mac0015027","name":"zap","msg":"Can't locate templates file: /Users/paregan/zap/error"}
{"level":40,"time":1680021934992,"pid":75175,"hostname":"mac0015027","name":"zap","msg":"Can't locate templates file: /Users/paregan/zap/error"}
{"level":30,"time":1680021935700,"pid":75175,"hostname":"mac0015027","name":"zap","msg":"Init message received: WebSocket initialized handshake.. Responding."}
{"level":50,"time":1680021935961,"pid":75175,"hostname":"mac0015027","name":"zap","msg":{}}**

**Comment by jingteng25742** [comment]
@cheng5230 
Are you running this on a headless setup? 
"zap" requires a display to setup correctly. 
If you are using headless mode / on a server, "zap-cli" will fit your needs better.
